cl/persistence/state/static_validator_table.go

```diff
@@ -230,7 +275,7 @@ func (s *StaticValidatorTable) AddValidator(v solid.Validator, validatorIndex, s
 		return nil
 	}
 	s.validatorTable = append(s.validatorTable, NewStaticValidatorFromValidator(v, slot))
-	if validatorIndex >= uint64(len(s.validatorTable)) {
+	if validatorIndex != uint64(len(s.validatorTable))-1 {
```

Comment:
I feel a bit confused about `validatorIndex`. 
if it must be equal to length of validatorTable here, then it's reasonable to remove it from function parameter.
checked code, seems `(b *BeaconState) AddValidator(validator solid.Validator, balance uint64)` is the only entry to this function.